libavfilter/sf_text2bitmap.c

```diff
@@ -0,0 +1,539 @@
+        [1] = AV_LOG_ERROR,     /* MSGL_ERR */
+        [2] = AV_LOG_WARNING,   /* MSGL_WARN */
+        [3] = AV_LOG_WARNING,   /* <undefined> */
+        [4] = AV_LOG_INFO,      /* MSGL_INFO */
+        [5] = AV_LOG_INFO,      /* <undefined> */
+        [6] = AV_LOG_VERBOSE,   /* MSGL_V */
+        [7] = AV_LOG_DEBUG,     /* MSGL_DBG2 */
+};
+
+static void ass_log(int ass_level, const char *fmt, va_list args, void *ctx)
+{
+    const int ass_level_clip = av_clip(ass_level, 0,
+        FF_ARRAY_ELEMS(ass_libavfilter_log_level_map) - 1);
+    const int level = ass_libavfilter_log_level_map[ass_level_clip];
+    av_vlog(NULL, level, fmt, args);
+    av_log(NULL, level, "\n");
+}
+
+static const char * const font_mimetypes[] = {
+        "font/ttf",
+        "font/otf",
+        "font/sfnt",
+        "font/woff",
+        "font/woff2",
+        "application/font-sfnt",
+        "application/font-woff",
+        "application/x-truetype-font",
+        "application/vnd.ms-opentype",
+        "application/x-font-ttf",
+        NULL
+};
+
+static int stream_is_font(AVStream* st)
+{
+    const AVDictionaryEntry *tag = NULL;
+    int n;
+
+    if (st->codecpar->codec_type != AVMEDIA_TYPE_ATTACHMENT)
+        return 0;
+
+    tag = av_dict_get(st->metadata, "mimetype", NULL, AV_DICT_MATCH_CASE);
+
+    if (tag) {
+        for (n = 0; font_mimetypes[n]; n++) {
+            if (av_strcasecmp(font_mimetypes[n], tag->value) == 0)
+                return 1;
+        }
+    }
+    return 0;
+}
+
+// Filter structures
+
+#define OFFSET(x) offsetof(ASS_Context, x)
+#define FLAGS (AV_OPT_FLAG_SUBTITLE_PARAM|AV_OPT_FLAG_FILTERING_PARAM)
+
+static const AVOption text2bitmap_options[] = {
+        { "w",          "output width",            OFFSET(width),      AV_OPT_TYPE_INT,    .min = 0, .max = 10000, .default_val.i64 = 0,    .flags = FLAGS },
+        { "width",      "output width",            OFFSET(width),      AV_OPT_TYPE_INT,    .min = 0, .max = 10000, .default_val.i64 = 0,    .flags = FLAGS },
+        { "h",          "output height",           OFFSET(height),     AV_OPT_TYPE_INT,    .min = 0, .max = 10000, .default_val.i64 = 0,    .flags = FLAGS },
+        { "height",     "output height",           OFFSET(height),     AV_OPT_TYPE_INT,    .min = 0, .max = 10000, .default_val.i64 = 0,    .flags = FLAGS },
+        { "n",          "number of output colors", OFFSET(num_colors), AV_OPT_TYPE_INT,    .min = 2, .max = 256,   .default_val.i64 = 256,  .flags = FLAGS },
+        { "num_colors", "number of output colors", OFFSET(num_colors), AV_OPT_TYPE_INT,    .min = 2, .max = 256,   .default_val.i64 = 256,  .flags = FLAGS },
+        { "f",          "fonts container",         OFFSET(fonts),      AV_OPT_TYPE_STRING,                         .default_val.str = NULL, .flags = FLAGS },
+        { "fonts",      "fonts container",         OFFSET(fonts),      AV_OPT_TYPE_STRING,                         .default_val.str = NULL, .flags = FLAGS },
```

Comment:
The naming is a bit weird. Nobody knows what a "fonts container" is. Also 'fonts' doesn't sound like you would need to specify a filename to media file containing font attachments.
I would just just copy the options from vf_subtitles:

https://github.com/softworkz/FFmpeg/blob/34aee50a7f5a4b36f9f7daf827422d345c7868ed/libavfilter/vf_subtitles.c#L69-L72

and change the description `set the filename of file to read` to something like `media file to load fonts from`